DOCS-5203 Environment Limitations

paigehargrave opened the pull request.

Commit:
Initial content

Commit:
Added new file to nav

paigehargrave commented:
Suggested updates for https://www.mulesoft.org/jira/browse/DOCS-5203.

federicoboerr commented on modules/ROOT/nav.adoc:
```diff
@@ -34,4 +35,4 @@
  ** xref:ext-analytics-elk.adoc[Analyze Business and API Data using ELK]
-* xref:upgrade.adoc[Migrating Anypoint Platform Private Cloud]
\ No newline at end of file
+* xref:upgrade.adoc[Migrating Anypoint Platform Private Cloud]
```
Is this part of this PR?

paigehargrave replied:
This was there before - not sure why it got flagged as a change.

federicoboerr commented on modules/ROOT/nav.adoc:
```diff
@@ -9,6 +9,7 @@
  ** xref:prereq-verify-device.adoc[Verify Device Requirements]
  ** xref:prereq-network.adoc[Network Requirements]
  ** xref:prereq-software.adoc[About Software Requirements]
+ ** xref:prereq-env.adoc[Environment Limitations]
```
Should this be included here or in the release notes? I'm not sure about the pros and cons.

paigehargrave replied:
I think it should be included here, unless it's something that's going to be fixed. Putting this in the release notes would mean classifying it as a 'Known Issue', which implies that it should be fixed. This information looks like a long term situation, which means it should go in the docs. Thoughts?

federicoboerr reviewed (commented):
some comments

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+include::_attributes.adoc[]
+endif::[]
+
+Adhere to the follow recommendations and limitations when setting up your environment and deploying apps:
```
following recommendations

paigehargrave replied:
Got it - thank you!

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+*** Applications per Environment 
+
+**** Less than 200 
```
Less > Fewer?

paigehargrave replied:
Changed - thank you!

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+* Environment:
+
+** Production: 6 Nodes of 8 cores and 32GB 
```
Should nodes be lowercase?

paigehargrave replied:
Yup - fixed.

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+**** Less than 200 
+
+***** Maximum: 500 Applications in a single environment 
```
Should Applications be lowercase (and maybe "apps"?) 

Global: look at capitalization and apply Kim's rules  :)

paigehargrave replied:
Got it - thanks!

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+***** Maximum: 500 Applications in a single environment 
+**** Number of Clusters/Runtimes
+
+***** Total: 800 runtimes 2 nodes each with 400 Clusters 
```
Kim has been changing my numerals to words: 2 > two.

paigehargrave replied:
OK - thanks! Got it.

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+***** Production: Maximum of 50 apps 
+***** Development: Maximum of 100 apps
+***** Recommended: Less than 25 apps
```
Fewer

paigehargrave replied:
Got it - thanks!

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+**** Number of Clusters/Runtimes
+
+***** Total: 800 runtimes 2 nodes each with 400 Clusters 
+**** Applications per Mule Runtime or Cluster
```
Kim says "Mule" or "Mule runtime engine"

paigehargrave replied:
OK - thanks! Got it.

hannanelson commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+***** A single API implementation with one policy counts as 0.3.
+***** A single clustered proxy with up to 5 policies counts as 1.
+***** Every runtime is polling every 60 seconds. 
```
All runtime (engines?) poll every 60 seconds?

paigehargrave replied:
Yup - clarified this. Thanks!

hannanelson reviewed (approved):
Some teeny edits.

Commit:
Updates per feedback

federicoboerr reviewed (approved):
LGTM

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,39 @@
+
+* Sizing
+
+** ARM:
```
Runtime Manager

Commit:
Combining info from dup PR

Info from https://github.com/mulesoft/docs-private-cloud/pull/52/files consolidated here.

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,42 @@
+
+* Environment:
+
+** Production: 6 nodes of 8 cores and 32GB 
```
and 32GB `RAM`

paigehargrave replied:
Got it - thank you!

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,42 @@
+* Environment:
+
+** Production: 6 nodes of 8 cores and 32GB 
+** Maximum number of servers: 500
```
All these belong below under `Runtime Management`

paigehargrave replied:
Moved - thank you!

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,42 @@
+
+* Sizing
+
+** Runtime Manager:
```
change to `Runtime Management`

paigehargrave replied:
Got it - thanks!

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,42 @@
+
+** Runtime Manager:
+
+*** Applications per Environment 
```
I'd say, merging with the text below as well:
```
*** Mule applications under management:
**** Recommended: around 200
**** Maximum: 500

*** Mule runtimes under management:
**** Maximum: 800 single Mule runtimes (or 400 Mule runtime clusters, consisting of 2 Mule runtimes per cluster)

*** Mule applications per Mule runtime:
***** Production: Maximum of 50 apps 
***** Development: Maximum of 100 apps
***** Recommended: Fewer than 25 apps
```

paigehargrave replied:
Made updates.

federicoboerr commented on modules/ROOT/pages/prereq-env.adoc:
```diff
@@ -0,0 +1,42 @@
+***** Recommended: Fewer than 25 apps
+** API Platform
+
+*** APIs under management
```
Add: 
```
** API Management
*** APIs under management
```

paigehargrave replied:
Got it - thanks!

federicoboerr reviewed (changes requested):
some re-ordering

Commit:
Updates per Fed

paigehargrave commented:
@federicoboerr Thank you!
